src/api/apiHelpers.ts

```diff
@@ -81,20 +81,19 @@ export const safeFetch = async (
   const token = authorizationHeader.replace('Bearer ', '')
   const accessTokenStatus = getTokenStatus(token)
 
-  if (accessTokenStatus === 'unknown') {
-    return createNeedsAuthenticationResponse(url)
-  }
-
-  // If the token is expired, we refresh it before calling the backend
-  if (accessTokenStatus === 'expired') {
+  // If the token is expired or unknown, we refresh it before calling the backend
+  if (accessTokenStatus === 'expired' || accessTokenStatus === 'unknown') {
     try {
       const { result: newAccessToken, error } = await refreshAccessToken(api)
+
       if (error === REFRESH_TOKEN_IS_EXPIRED_ERROR) {
         return RefreshTokenExpiredResponse
       }
 
       if (error) {
-        eventMonitoring.captureException(new Error(`safeFetch ${error}`))
+        eventMonitoring.captureException(new Error(`safeFetch ${error}`), {
```

Comment:
je pense que ça pourrait etre bien de conserver la callstack

```suggestion
        eventMonitoring.captureException(new Error(`safeFetch ${error}`, {cause:error}), {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

Reply:
pareil en dessous

Reply:
On risque de remonter beaucoup de "fausses" erreurs :
- J'arrive sur l'app pour la première fois (ou juste sans être connecté) sur une URL qui demande d'être authentifier
- Je n'ai pas d'access token (status `unknown`) ni de refresh token
=> Je remonte une erreur à Sentry (level `error` et pas info)